File: 알고리즘/2/Goal Parser Interpretation.cpp
class Solution {
public:
    string interpret(string command) {
        stack<char> st;
        string result;
        for(auto ele : command){
            if(ele=='G'){
                result+=ele;
                continue;
            }

            if(ele=='('){
                st.push(ele);
                continue;
            }

            if(st.top()=='(' && ele == ')'){
                st.pop();
                result+='o';
                continue;
            }

            if(st.top()=='l'&& ele==')'){
                st.pop();
                st.pop();
                st.pop();
                result+='al';
                continue;
            }

            st.push(ele);
        }
    }

    string interpret(string command) {
        replace(command.begin(), command.end(), "()", "o");
        replace(command.begin(), command.end(), "(al)", "al");
        return command
    }
};

File: 알고리즘/2/Goal Parser Interpretation.py
class Solution:
    def interpret(self, command: str) -> str:
        stack =[]
        result =''
        for ele in command:
            if ele=='G':
                result+='G'
                continue
            
            if ele=='(':
                stack.append('(')
                continue
            
            if stack[-1] == '(' and ele==')':
                result+='o'
                continue

            if stack[-1]=='l' and ele ==')':
                result+='al'
                stack.pop()
                stack.pop()
                stack.pop()
                continue

            stack.append(ele)
        
        return result
                